test/test_torch.py

```diff
@@ -3319,6 +3319,30 @@ def rand_tensor(size, dtype, device):
             out = torch.addcmul(a, b, c, value=-1)
             self.assertTrue(not (out.isnan() or out.isinf()))
 
+    def test_addcmul_cpu_scalar_errors(self, device):
+        # Logic is dtype agnostic, so dtype isn't tested
+        alpha = 0.5
+
+        a = torch.rand((2, 2), device=device)
+        b = torch.rand((2, 2), device=device)
+        c = torch.rand((2, 2), device=device)
+        scalar = torch.rand([], device="cpu")
+
+        torch.addcmul(a, b, c, value=alpha)
+        torch.addcmul(a, b, scalar, value=alpha)  # supported on both
+
+        if device == "cpu":
+            torch.addcmul(a, scalar, c, value=alpha)
+            torch.addcmul(scalar, b, c, value=alpha)
```

Comment:
It would be better to assert that these equal'd a reference impl vs just not erroring